### PR TITLE
Add CCPA us_privacy support to spotxBidAdapter

### PR DESCRIPTION
**Pull request by npeceniak**
## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Add CCPA us_privacy support to spotxBidAdapter

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

https://github.com/prebid/prebid.github.io/pull/1727

### CODE REVIEW

File: modules/spotxBidAdapter.js [hunk 1]
```diff
@@ -202,14 +202,16 @@ export const spec = {
         userExt.consent = bidderRequest.gdprConsent.consentString;
 
         if (typeof bidderRequest.gdprConsent.gdprApplies !== 'undefined') {
-          requestPayload.regs = {
-            ext: {
-              gdpr: (bidderRequest.gdprConsent.gdprApplies ? 1 : 0)
-            }
-          };
+          requestPayload.regs = requestPayload.regs || {ext: {}};
```
Comment (msm0504) [inline modules/spotxBidAdapter.js]: This isn't needed.
`utils.deepSetValue` can be used to set `regs.ext.gdpr` [docs](https://www.npmjs.com/package/dset)

File: modules/spotxBidAdapter.js [hunk 2]
```diff
@@ -202,14 +202,16 @@ export const spec = {
...
+          requestPayload.regs.ext.gdpr = (bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
         }
       }
 
+      if (bidderRequest && bidderRequest.uspConsent) {
+        requestPayload.regs = requestPayload.regs || {ext: {}};
```
Comment (msm0504) [inline modules/spotxBidAdapter.js]: This isn't needed.
`utils.deepSetValue` can be used to set `regs.ext.us_privacy` [docs](https://www.npmjs.com/package/dset)

File: modules/spotxBidAdapter.js (hunk 1 above)
Comment (npeceniak) [reply modules/spotxBidAdapter.js]: Update to use utils.deepSetValue

File: modules/spotxBidAdapter.js (hunk 2 above)
Comment (npeceniak) [reply modules/spotxBidAdapter.js]: Update to use `utils.deepSetValue`

**Pull request merged**